File: TrainReservation.Core/GraphQL/Types/SeatType.cs
using HotChocolate.Types;
using TrainReservation.Core.Models;

namespace TrainReservation.Core.GraphQL.Types
{
    public class SeatType : ObjectType<Seat>
    {
        protected override void Configure(IObjectTypeDescriptor<Seat> descriptor)
        {
            descriptor.Name("Seat");
            descriptor.Description("Represents a seat within a train car, including its reservations and availability.");

            descriptor.Field(f => f.Id).Type<IntType>().Description("The unique identifier of the seat.");
            descriptor.Field(f => f.Number).Type<IntType>().Description("The seat number within the car.");
            descriptor.Field(f => f.Car).Type<CarType>().Description("The car to which this seat belongs.");
            descriptor.Field(f => f.Reservations).Type<ListType<ReservationType>>().Description("The list of reservations associated with this seat.");
            descriptor.Field(f => f.Calendars).Type<ListType<CalendarType>>().Description("The list of calendar entries relevant to this seat.");
            descriptor.Field(f => f.SeatCalendars).Type<ListType<SeatCalendarType>>().Description("The list of seat-calendar availability mappings.");
            descriptor.Field(f => f.ReservationSeats).Type<ListType<ReservationSeatType>>().Description("The list of reservation-seat mappings for this seat.");
        }
    }
}

File: TrainReservation.UI/src/app/components/trains/graphql/types/seat-type.ts
import { Car } from "./car-type";
import { SeatCalendar } from "./seat-calendar-type";
import { Calendar } from "./calendar-type";
import { Reservation } from "../../../reservations/graphql/types/reservation-type";
import { ReservationSeat } from "../../../reservations/graphql/types/reservation-seat-type";


export type Seat = {
    id: number;
    number: number;
    car: Car;
    reservations: Reservation[];
    calendars: Calendar[];
    seatCalendars: SeatCalendar[];
    reservationSeats: ReservationSeat[];
};
